### PR TITLE
feat: add hot reloading on dev mode

### PR DESCRIPTION
**Pull request by zulkhair**
Closes: WORLD-757

## Overview
  
Adding hot reload on `world cardinal dev` command

## Brief Changelog
  
- Modify a way to run cardinal from using plain `go run` to wrapped using `fresh` library

## Testing and Verifying

- Run the binary in linux(ubuntu) 
- Run the binary in windows 

## Summary by CodeRabbit

- **Refactor**
	- Enhanced development environment setup for better efficiency.
- **Chores**
	- Integrated a subproject update to ensure the latest features and fixes are included.
- **New Features**
	- Added import for `github.com/gravityblast/fresh/runner`.
- **Bug Fixes**
	- Removed the `runCardinal` function call and related error handling.
	- Changed directory to "cardinal" and set environment variable "RUNNER_IGNORED" before starting `runner.Start()`.

### CODE REVIEW

**Comment by zulkhair** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #33** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/33?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Argus-Labs/world-cli/33?utm_source=stack-comment).

File: cmd/world/cardinal/dev.go [hunk 1]
```diff
@@ -65,23 +66,19 @@ var devCmd = &cobra.Command{
...
+		if err != nil {
+			return err
+		}
 
 		fmt.Println("Starting Cardinal...")
 		go func() {
-			err := cardinalExecCmd.Wait()
-			cmdErr <- err
+			os.Setenv("RUNNER_IGNORED", "assets, tmp, vendor")
+			runner.Start()
```
[the remark this answers is not in the document]
Comment (zulkhair) [reply cmd/world/cardinal/dev.go:81]: runner.Start() didn't return error

File: cmd/world/cardinal/dev.go [hunk 2]
```diff
@@ -65,62 +69,36 @@ var devCmd = &cobra.Command{
...
-		// Create a channel to receive errors from the command
-		cmdErr := make(chan error, 1)
+		// Run Cardinal Preparation
+		err = runCardinalPrep()
+		if err != nil {
+			return err
+		}
 
 		fmt.Println("Starting Cardinal...")
+		go func(stopChan <-chan struct{}) {
```
Comment (jerargus) [inline cmd/world/cardinal/dev.go:83]: nit: stopChan seems to be unused. Wouldn't this be the same:

go runner.Start()

**Review by jerargus — APPROVED** [review]

**Comment by jerargus** [comment]
Holy cow, this is great!

File: cmd/world/cardinal/dev.go (hunk 2 above)
Comment (zulkhair) [reply cmd/world/cardinal/dev.go:83]: Hmm after i test it again, yup that's right, its unused.
My purpose is, i want to stop the go routine process when i'm closing the channel (for unit test purpose).

i'll be back with another method.

Comment (zulkhair) [reply cmd/world/cardinal/dev.go]: Updated

**Pull request merged**